kidneys_misc.module

```diff
@@ -61,49 +66,214 @@ function kidneys_misc_form_alter(&$form, &$form_state, $form_id) {
+//      break;
+  }
+
+  // Pass values to the form.
+  $form['fields_to_keep'] = $fields_to_keep;
+  $form['fields_true'] = $fields_true;
+  $form['fields_false'] = $fields_false;
+  $form['address_field'] = $address_field;
+  $form['trial_redirect_path'] = $trial_redirect_path;
+  $form['trial_location_field_name'] = $trial_location_field_name;
+
+  // Process trial form data:
+  kidneys_misc_process_trial_form($form, $form_state);
+}
+
+/**
+ * Process submitted trial form data.
+ *   - Process true and false values.
+ *   - Process address and get closest trial location.
+ *   - Clear out unwanted field data.
+ *
+ * @param $form
+ * @param $form_state
+ */
+function kidneys_misc_process_trial_form($form, &$form_state) {
+  $access = FALSE;
+  $location = $address_string = '';
+
+  if (isset($form_state['entityform'])) {
+    $wrapper = entity_metadata_wrapper('entityform',$form_state['entityform']);
+
+    // Check for TRUE values.
+    if (empty($form['fields_true'])) {
+      $true_passes = FALSE;
+    }
+    else {
+      $true_passes = TRUE;
+    }
+    // If any fields are false, true test doesn't pass.
+    foreach ($form['fields_true'] as $true_field) {
+      if (isset($wrapper->{$true_field})) {
+        if (empty($wrapper->{$true_field}->value())) {
+          $true_passes = FALSE;
+        }
+      }
+    }
+
+    // Check for FALSE values.
+    if (empty($form['fields_false'])) {
+      $false_passes = FALSE;
+    }
+    else {
+      $false_passes = TRUE;
+    }
+    // If any fields are true, false test doesn't pass.
+    foreach ($form['fields_false'] as $false_field) {
+      if (isset($wrapper->{$false_field})) {
+        if (!empty($wrapper->{$false_field}->value())) {
+          $false_passes = FALSE;
+        }
+      }
+    }
+
+    // If TRUE and FALSE values are as expected set access to TRUE.
+    if ($true_passes && $false_passes) {
+      $access = TRUE;
+    }
+
+    // Process address.
+    if (isset($wrapper->{$form['address_field']})) {
+      if (!empty($address = $wrapper->{$form['address_field']}->value())) {
+        $address_string = $address['locality'];
+        $address_string .= ', ' . $address['administrative_area'];
+        $address_string .= ', ' . $address['postal_code'];
+      }
+    }
+
+    // If access is allowed and there is an address
+    if ($access && $address_string) {
+      // Get the entityform_type id for views filtering.
+      $bundle = $wrapper->getBundle();
+      $query = "SELECT id FROM {entityform_type} WHERE type = :type";
+      $entity_form_type_id = db_query($query, array(':type' => $bundle))->fetchField();
+
+      // Geocode the address string.
+      $point = geocoder('google', $address_string);
       if (!empty($point)) {
         $lon = $point->coords[0];
         $lat = $point->coords[1];
-        $arg = $lat . ',' . $lon;
+        $geo_coordinates = $lat . ',' . $lon;
         // Use view to look up location, this makes it easy to adjust in UI.
```

Comment:
Since it's different than how it was originally built, let's add a bit more to this comment about how the view uses the entity form bundle to find locations associated with that entity form type.

Reply:
Good call.